### PR TITLE
Rename "ready" task state to "preparing".

### PR DESCRIPTION
**Pull request by hjoliver**
UI counterpart of https://github.com/cylc/cylc-flow/pull/4014

This is a small change with no associated Issue.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: src/components/cylc/gscan/index.js [hunk 1]
```diff
@@ -31,10 +31,6 @@ function getWorkflowSummary (workflow) {
       // a task in waiting, may not have any jobs
       if (taskProxy.jobs) {
         for (const job of taskProxy.jobs) {
-          // TODO: temporary fix, as the backend is sending ready jobs, but they will change in cylc flow&uiserver in the future
-          if (job.state === 'ready') {
-            continue
-          }
```
Comment (hjoliver) [inline src/components/cylc/gscan/index.js:37]: (Not sure what this block was for?)

Comment (kinow) [reply src/components/cylc/gscan/index.js:37]: I think we had jobs with state `ready`, but only tasks could have this state? I remember asking something about that in Element or GH, and then was told that that was not correct for jobs (if I recall correctly), and that there was a ticket somewhere about improvements for states...

I think this is not preventing a JavaScript error, but preventing `Job` components of being created with the `state=ready` in the UI.

Comment (hjoliver) [reply src/components/cylc/gscan/index.js:37]: Ah, OK, we'll have to look to see if that is still a problem (which now will be, are there jobs with the "preparing" state, which is equally wrong).

**Review by kinow — APPROVED** [review]
Probably needs the Cylc Flow PR merged first, but the code here looks good already :+1: 

Tests passed, and coverage even increased a bit (code removed) https://app.codecov.io/gh/cylc/cylc-ui/compare/564

+1 pre-approving, leaving for a second reviewer to review and/or merge later

**Review by oliver-sanders — APPROVED** [review]

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**